src/lib/add-matching-cards-to-results.ts

```diff
@@ -1,19 +1,22 @@
-'use strict';
+import clone from './clone';
+import matches from './matches';
+import { CreditCardType } from '../index';
 
-var clone = require('./clone');
-var matches = require('./matches');
-
-function addMatchingCardsToResults(cardNumber, cardConfiguration, results) {
-  var i, pattern, patternLength, clonedCardConfiguration;
+export default function addMatchingCardsToResults(
+  cardNumber: string,
+  cardConfiguration: CreditCardType,
+  results: Array<object>
+): void {
+  let i, patternLength;
 
   for (i = 0; i < cardConfiguration.patterns.length; i++) {
-    pattern = cardConfiguration.patterns[i];
+    const pattern = cardConfiguration.patterns[i];
```

Comment:
i'm shocked neither ts nor ESLint are yelling at us for declaring a const inside a loop

Reply:
Yeah, I thought that was weird too.

Reply:
I believe the reason this is fine is because `const` are block scoped.

It wouldn't work if we did:

```
for (const i = 0;
```

But declaring a variable inside the block of the loop is fine.

Reply:
oh i wasn't meaning like scope-related errors, i just meant efficiency, speed, & code style stuff